base/redactor_userdata.go

```diff
@@ -56,6 +56,10 @@ func UD(i interface{}) RedactorFunc {
 		return func() Redactor {
 			return UserData(v)
 		}
+	case Set:
+		return func() Redactor {
+			return UD(v.ToArray())
```

Comment:
This approach looks like it's going to mean three iterations over the dataset (once in ToArray, once in the buildRedactorFuncSlice, and once in RedactorSlice.Redact).  It also has whatever overhead is associated with allocation of the intermediate slices, as well as the otherwise unnecessary reflection work.   Given that, and that set logging may end up on performance-critical paths, is there an option to have Set implement Redactor?

Reply:
Have modified the approach to build a RedactorSlice using the new function set.buildRedactor(). This works in a similar way to the slice work but cuts out some of the work. It removes one iteration and removes the reflection work. It still has two iterations but I think this is the best way forward. Has an interation to build the redactor and then another when the item is either redacted or not redacted.